src/manifold/src/manifold.cpp

```diff
@@ -517,6 +517,30 @@ Manifold Manifold::Transform(const glm::mat4x3& m) const {
   return Manifold(pNode_->Transform(m));
 }
 
+/**
+ * Mirror this Manifold over the plane described by the unit form of the given
+ * normal vector. If the length of the normal is zero, an empty Manifold is
+ * returned. This operation can be chained. Transforms are combined and applied
+ * lazily.
+ *
+ * @param normal The normal vector of the plane to be mirrored over
+ */
+Manifold Manifold::Mirror(glm::vec3 normal) const {
+  if (glm::length(normal) == 0.) {
+    return Manifold();
+  }
+  auto n = glm::normalize(normal);
+  glm::mat4x3 m(1.0f - 2.0f * n.x * n.x, -2.0f * n.x * n.y,
+                -2.0f * n.x * n.z,  //
+                -2.0f * n.x * n.y, 1.0f - 2.0f * n.y * n.y,
+                -2.0f * n.y * n.z,  //
+                -2.0f * n.x * n.z, -2.0f * n.y * n.z,
+                1.0f - 2.0f * n.z * n.z  //
+                ,
+                0.0f, 0.0f, 0.0f);
```

Comment:
Could this be rewritten as identity - outerProduct? Like `glm::mat4x3(glm::mat3(1) - glm::outerProduct(normal, normal))`.

Reply:
I just did it this "manual" way because that is what I have used in the past based on other libraries and math notes. That does look like it describes the same operation though.

Reply:
Actually, no seems like not. Your equation worked with `{1, 1, 0}` as in current test, but a normal of `{1, 1, 1}` it gives this broken output:
![image](https://user-images.githubusercontent.com/14910691/223634870-28c14e0f-08d7-48bb-b56b-4bad4b0a176d.png)
When the other way gives this:
![image](https://user-images.githubusercontent.com/14910691/223635502-0f3f770f-0b71-4825-943a-45c8f2f9710a.png)
Which appears correct.

Reply:
Oh, I think I missed the factor of 2: `glm::mat4x3(glm::mat3(1) - 2.0 * glm::outerProduct(normal, normal))`

Reply:
Ah yes, that works (with a normalized normal).